### PR TITLE
Implement new rent notification mail

### PR DESCRIPTION
**Pull request by sebasdeldi**
## Summary
- Sends notification email when a rent is created
- Implements a queue system through sidekiq and redis
## Trello Card
https://trello.com/c/GbgdtZIA/10-mandar-email-al-usuario-cuando-se-crea-un-rent

### CODE REVIEW

File: app/mailers/api/v1/rents_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module Api
+  module V1
+    class RentsMailer < ApplicationMailer
+      def new_rent(rent)
```
Comment (Gmarn) [inline app/mailers/api/v1/rents_mailer.rb]: It's better when you use redis as intermediate to no pass complete model. Prefer passing the rent_id and do a find to found it ;)

File: app/controllers/api/v1/rents_controller.rb [hunk 2]
```diff
@@ -5,11 +5,12 @@ class RentsController < ApiController
 
       def create
         rent = Rent.create(rent_params)
+        Api::V1::RentsMailer.new_rent(rent).deliver_later
```
Comment (Gmarn) [inline app/controllers/api/v1/rents_controller.rb]: This mail should be send only if the creation is a success. But you are implementing this in the previous PR ;)

File: app/mailers/api/v1/rents_mailer.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+module Api
+  module V1
+    class RentsMailer < ApplicationMailer
+      def new_rent(rent_id)
+        @rent = Rent.find(rent_id)
+        @user = @rent.user
+        @book = @rent.book
+        mail to: @user.email
```
Comment (nicozar95) [inline app/mailers/api/v1/rents_mailer.rb:8]: We could add a subject like `subject: 'New rent information'` but is no biggi

File: config/initializers/action_mailer.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+Rails.application.configure do
+  config.action_mailer.delivery_method = :smtp
+
+  config.action_mailer.smtp_settings = {
+    user_name: Rails.application.secrets.mailer_username,
+    password: Rails.application.secrets.mailer_password,
+    address: Rails.application.secrets.mailer_address,
+    domain: Rails.application.secrets.mailer_domain,
+    port: '2525',
```
Comment (nicozar95) [inline config/initializers/action_mailer.rb]: Thought this is ok, we could move it to an ENV variable if in the future we need to change some of the configuration

**Review by nicozar95 — APPROVED** [review]

**Pull request merged**